### PR TITLE
refactor: rename `configs` to `defaults`, remove `swarm`

### LINKED ISSUE #4293: [discussion] Rename `configs/swarm`?
After #4239, docker-compose configs (YAML files and related Iroha configs) are placed into the `configs/swarm` directory. @mversic suggested that renaming it to `configs/docker` might be clearer. I would go further and suggest naming it `configs/docker_compose`, because _Docker_ configs (i.e. Dockerfiles) are still in the repo root.

### PR DESCRIPTION
**Pull request by nxsaken**
## Description

Rename the `configs` directory to `defaults` since that's what it really is (default docker compose configs, default `executor.wasm`, default `genesis.json`, etc.)

### Linked issue

Closes #4293, #4320

### Benefits

Conveys the purpose of the directory better.

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [x] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

**Comment by mversic** [comment]
This was renamed for the 5th time already. We have to agree on this for once

File: defaults/client.template.toml
Comment (0x009922) [inline defaults/client.template.toml:1]: <kbd>defaults</kbd> for default configs makes sense to me, but it doesn't make much sense for `*.template.*` files. Shall we maybe move them to <kbd>docs/source/references</kbd>?

**Review by 0x009922 — APPROVED** [review]

File: torii/pytests/common/settings.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 BASE_DIR = os.path.dirname(
     os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 )
-CONFIG_DIR = os.path.join(BASE_DIR, "configs/swarm/client.toml")
+CONFIG_DIR = os.path.join(BASE_DIR, "defaults/client.toml")
```
Comment (mversic) [inline torii/pytests/common/settings.py:10]:
```suggestion
CONFIG_DIR = os.path.join(BASE_DIR, "default/client.toml")
```

I'd prefer singular?

**Review by mversic — DISMISSED** [review]

File: torii/pytests/common/settings.py (hunk 1 above)
Comment (nxsaken) [reply torii/pytests/common/settings.py:10]: I like the plural version more, but I can't put my finger on why

**Review by mversic — APPROVED** [review]

**Review by 0x009922 — APPROVED** [review]

**Pull request merged**